src/wallet/wallet.cpp

```diff
@@ -2602,94 +2602,111 @@ std::shared_ptr<CWallet> CWallet::Create(interfaces::Chain* chain, const std::st
     }
 
     if (gArgs.IsArgSet("-mintxfee")) {
-        CAmount n = 0;
-        if (!ParseMoney(gArgs.GetArg("-mintxfee", ""), n) || 0 == n) {
-            error = AmountErrMsg("mintxfee", gArgs.GetArg("-mintxfee", ""));
-            return nullptr;
-        }
-        if (n > HIGH_TX_FEE_PER_KB) {
+        if (std::optional<CAmount> min_tx_fee = ParseMoney(gArgs.GetArg("-mintxfee", ""))) {
+            if(min_tx_fee.value() == 0) {
```

Comment:
```suggestion
            if (min_tx_fee.value() == 0) {
```

Reply:
Also wouldn't it be easier to review to keep the `min_tx_fee` symbol outside the if and then keep the if conditions the same?

```cpp
fee = Parse;
if (!fee || !*fee) {
 // same
}
// same

Reply:
Possibly, going to leave for the minute, as I could change this again in a followup.